libfwupdplugin/fu-device.c

```diff
@@ -2968,6 +2972,12 @@ fu_device_add_flag(FuDevice *self, FwupdDeviceFlags flag)
 	if (flag & FWUPD_DEVICE_FLAG_IS_BOOTLOADER)
 		fu_device_remove_flag(self, FWUPD_DEVICE_FLAG_NEEDS_BOOTLOADER);
 
+	/* being both a signed and unsigned is invalid */
+	if (flag & FWUPD_DEVICE_FLAG_SIGNED_PAYLOAD)
+		fu_device_remove_flag(self, FWUPD_DEVICE_FLAG_UNSIGNED_PAYLOAD);
+	if (flag & FWUPD_DEVICE_FLAG_UNSIGNED_PAYLOAD)
+		fu_device_remove_flag(self, FWUPD_DEVICE_FLAG_SIGNED_PAYLOAD);
```

Comment:
I don't like this - it's a deterministic failure order and may mask a problem.  Can you also a warning?

Reply:
Yes, I was worried about that too; but I think it's being a bit pragmatic. For instance the hundreds of PixArt devices are all unsigned, with the exception of one vendor design. In this case I think it makes sense to set unsigned in the .c code and then just "override" it for that one VID/PID in the quirk file.